homeassistant/components/cover/garadget.py

```diff
@@ -0,0 +1,271 @@
+            "device_id": cover.get(CONF_DEVICE),
+            "username": cover.get(CONF_USERNAME),
+            "password": cover.get(CONF_PASSWORD),
+            "access_token": cover.get(CONF_ACCESS_TOKEN)
+        }
+
+        covers.append(GaradgetCover(hass, args))
+
+    add_devices(covers)
+
+
+class GaradgetCover(CoverDevice):
+    """Representation of a demo cover."""
+
+    # pylint: disable=no-self-use, too-many-instance-attributes
+    def __init__(self, hass, args):
+        """Initialize the cover."""
+        self.particle_url = 'https://api.particle.io'
+        self.hass = hass
+        self._name = args['name']
+        self.device_id = args['device_id']
+        self.access_token = args['access_token']
+        self.obtained_token = False
+        self._username = args['username']
+        self._password = args['password']
+        self._state = STATE_UNKNOWN
+        self.time_in_state = None
+        self.signal = None
+        self.sensor = None
+        self._unsub_listener_cover = None
+        self._hidden = False
+
+        if self.access_token is None:
+            self.access_token = self.get_token()
+            self._obtained_token = True
+
+        # Lets try to get the configured name if not provided.
+        try:
+            if self._name is None:
+                doorconfig = self._get_variable("doorConfig")
+                if doorconfig["nme"] is not None:
+                    self._name = doorconfig["nme"]
+            self.update()
+        except requests.exceptions.ConnectionError as ex:
+            _LOGGER.error('Unable to connect to server: %(reason)s',
+                          dict(reason=ex))
+            self._state = STATE_OFFLINE
+            self._hidden = True
+            self._name = DEFAULT_NAME
+        except KeyError as ex:
+            _LOGGER.warning('Garadget device %(device)s seems to be offline',
+                            dict(device=self.device_id))
+            self._name = DEFAULT_NAME
+            self._state = STATE_OFFLINE
+            self._hidden = True
+
+    def __del__(self):
+        """Try to remove token."""
+        if self._obtained_token is True:
+            if self.access_token is not None:
+                self.remove_token()
+
+    @property
+    def name(self):
+        """Return the name of the cover."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """No polling needed for a demo cover."""
+        return True
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        data = {}
+
+        if self.signal is not None:
+            data[ATTR_SIGNAL_STRENGTH] = self.signal
+
+        if self.time_in_state is not None:
+            data[ATTR_TIME_IN_STATE] = self.time_in_state
+
+        if self.sensor is not None:
+            data[ATTR_SENSOR_STRENGTH] = self.sensor
+
+        if self._hidden is not None:
+            data[ATTR_HIDDEN] = self._hidden
```

Comment:
Don't set this attribute. If you need to set it, use the `hidden` property. However, in your case it is not needed - use `available` instead.